### PR TITLE
Feature/create cache

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

File: src/UserService.Broker/Consumers/GetUserDataConsumer.cs [hunk 1]
```diff
@@ -5,40 +5,37 @@
...
+    private object GetUserData(IGetUserDataRequest request)
+    {
+      DbUser dbUser = _repository.Get(request.UserId);
 
-        public async Task Consume(ConsumeContext<IGetUserDataRequest> context)
-        {
-            var response = OperationResultWrapper.CreateResponse(GetUserData, context.Message);
+      if (dbUser == null)
+      {
+        throw new NotFoundException("The user was not found.");
```
Comment (Cheephot) [inline src/UserService.Broker/Consumers/GetUserDataConsumer.cs:22]: return null

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 2]
```diff
@@ -155,27 +255,25 @@ private IGetDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, in
...
-        var request = IGetCompanyEmployeesRequest.CreateObj(
-          usersIds,
-          includeDepartments,
-          includePositions,
-          includeOffices);
-
-        var response = _rcGetCompanyEmployees
-          .GetResponse<IOperationResult<IGetCompanyEmployeesResponse>>(request)
-          .Result;
+        var response = await _rcGetCompanyEmployees
```
Comment (Cheephot) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:262]: don't use var pls

**Review by Cheephot — APPROVED** [review]

File: src/UserService.Broker/Consumers/GetUserDataConsumer.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/UserService.Broker/Consumers/GetUserDataConsumer.cs:22]: in consumers we can use exceptions

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 3]
```diff
@@ -143,7 +150,100 @@ private IGetDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, in
       return null;
     }
 
-    private IGetCompanyEmployeesResponse GetCompanyEmployess(
+    private async Task<(List<DepartmentData> departments, List<PositionData> positions, List<OfficeData> offices)> GetCompanyEmployess(
+        List<Guid> usersIds,
```
Comment (VLiamin) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:154]: 4 spaces instead of 2

**Review by VLiamin — APPROVED** [review]

File: src/UserService.Business/Commands/User/FindUserCommand.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/FindUserCommand.cs]: fixed

File: src/UserService.Broker/Consumers/GetUserDataConsumer.cs [hunk 4]
```diff
@@ -5,40 +5,37 @@
...
+      _repository = repository;
+    }
 
-            if (dbUser == null)
-            {
-                throw new NotFoundException("The user was not found.");
-            }
+    public async Task Consume(ConsumeContext<IGetUserDataRequest> context)
+    {
+      var response = OperationResultWrapper.CreateResponse(GetUserData, context.Message);
```
Comment (Glunky) [inline src/UserService.Broker/Consumers/GetUserDataConsumer.cs:36]: var

File: src/UserService.Broker/Consumers/GetUsersDataConsumer.cs [hunk 5]
```diff
@@ -1,51 +1,73 @@
...
+  public class GetUsersDataConsumer : IConsumer<IGetUsersDataRequest>
+  {
+    private readonly IUserRepository _repository;
+    private readonly IConnectionMultiplexer _cache;
+    private readonly IOptions<RedisConfig> _redisConfig;
+
+    private List<UserData> GetUserInfo(IGetUsersDataRequest request)
     {
-        private readonly IUserRepository _repository;
+      var dbUsers = _repository.Get(request.UserIds);
```
Comment (Glunky) [inline src/UserService.Broker/Consumers/GetUsersDataConsumer.cs:32]: var

File: src/UserService.Business/Commands/User/FindUserCommand.cs [hunk 6]
```diff
@@ -115,7 +122,7 @@ private List<RoleData> GetRoles(List<Guid> userIds, List<string> errors)
     private IGetDepartmentUsersResponse GetUserIdsByDepartment(Guid departmentId, int skipCount, int takeCount, List<string> errors)
     {
       string errorMessage =
```
Comment (Glunky) [inline src/UserService.Business/Commands/User/FindUserCommand.cs:124]: used once - dont need variable

**Review by Glunky — APPROVED** [review]

File: src/UserService.Broker/Consumers/GetUserDataConsumer.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/UserService.Broker/Consumers/GetUserDataConsumer.cs]: fixed

File: src/UserService.Broker/Consumers/GetUsersDataConsumer.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/UserService.Broker/Consumers/GetUsersDataConsumer.cs]: fixed

File: src/UserService.Business/Commands/User/FindUserCommand.cs (hunk 6 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/FindUserCommand.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**